cognite_toolkit/_cdf_tk/loaders/_resource_loaders/auth_loaders.py

```diff
@@ -315,23 +316,61 @@ def dump_resource(self, resource: Group, local: dict[str, Any] | None = None) ->
         # When you dump a CDF Group, all the referenced resources should be available in CDF.
         return self._substitute_scope_ids(dumped, is_dry_run=False, reverse=True)
 
-    def _upsert(self, items: Sequence[GroupWrite]) -> GroupList:
+    def create(self, items: Sequence[GroupWrite]) -> GroupList:
         if len(items) == 0:
             return GroupList([])
+        return self.client.iam.groups.create(items)
+
+    def update(self, items: Sequence[GroupWrite]) -> GroupList:
         # We MUST retrieve all the old groups BEFORE we add the new, if not the new will be deleted
         old_groups = self.client.iam.groups.list(all=True)
-        old_group_by_names = {g.name: g for g in old_groups.as_write()}
-        to_create = []
-        for item in items:
-            if (old := old_group_by_names.get(item.name)) and old == item:
-                # Skip unchanged groups
-                continue
-            to_create.append(item)
-        if len(to_create) == 0:
-            return GroupList([])
-        created = self.client.iam.groups.create(to_create)
+        created = self.client.iam.groups.create(items)
         created_names = {g.name for g in created}
-        to_delete = [g.id for g in old_groups if g.name in created_names and g.id]
+        to_delete = GroupList([group for group in old_groups if group.name in created_names])
+        if to_delete:
+            self._delete(to_delete, check_own_principal=False)
```

Comment:
Both the former `_upsert` and the `delete` had the same logic so combined it.

Note it is allowed to delete a group your service principal is using as it has already been created first, so there is little risk in locking yourself out.